### PR TITLE
Refactor(theme answers): Consolidate logic into frontend [FS-4046]

### PR DESCRIPTION
**Pull request by tferns**
Migrated transformation logic for theme answers from the assessment store to the frontend. This centralization supports an upcoming major refactor, reducing complexity by simplifying future modifications. The assessment store now only returns necessary data, streamlining the architecture and making the refactor process more manageable. Functionality verified post-migration.

# Reasons
- Facilitates a cleaner, more efficient refactor.
- Centralizes logic for easier future updates.

# Changes
- Logic moved to frontend; assessment store simplified.

---

- [x] Unit tests and other appropriate tests added or updated
- [x] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines

### CODE REVIEW

File: tests/test_get_sub_criteria.py
[the remark this answers is not in the document]
Comment (tferns) [reply tests/test_get_sub_criteria.py:1]: Sure - should be easy enough to copy paste them over, although they won't live long I don't think 😆

[the remark this answers is not in the document]
Comment (tferns) [reply tests/test_get_sub_criteria.py:1]: https://github.com/communitiesuk/funding-service-design-assessment/pull/559/commits/fd82cbd4f46976a3b46abeabfa04a70efabd3f7f

**Comment by tferns** [comment]
Looking at why https://github.com/communitiesuk/funding-service-design-assessment-store/actions/runs/7869261998/job/21511152182?pr=408 is failing - re-running

**Comment by RamuniN** [comment]
> Looking at why https://github.com/communitiesuk/funding-service-design-assessment-store/actions/runs/7869261998/job/21511152182?pr=408 is failing - re-running

Had a similar error in fund-store. Rob is rolling out a fix to all the repos

**Review by RamuniN — DISMISSED** [review]

**Review by RamuniN — APPROVED** [review]

**Pull request merged**